app/src/main/java/org/sagebionetworks/research/mpower/ActivityRecorderService.kt

```diff
@@ -0,0 +1,280 @@
+    private var isRecording = false
+
+    @Inject
+    lateinit var taskResultManager: TaskResultManager
+
+    @Inject
+    lateinit var recorderConfigPresentationFactory: RecorderConfigPresentationFactory
+
+    private lateinit var recorderManager: RecorderManager
+
+    private lateinit var task: Task
+
+    private lateinit var taskIdentifier: String
+
+    private lateinit var taskUUID: UUID
+
+    private val asyncActions = mutableSetOf<AsyncActionConfiguration>(
+        DeviceMotionRecorderConfigurationImpl.builder()
+            .setIdentifier("motionRecorder")
+            .setStartStepIdentifier("start")
+            .setStopStepIdentifier("stop")
+            .setFrequency(null)
+            .setRecorderTypes(mutableSetOf(RecorderType.MOTION))
+            .build()
+//        DistanceRecorderConfigurationImpl.builder()
+//            .setIdentifier("distanceRecorder")
+//            .setStartStepIdentifier("start")
+//            .setStopStepIdentifier("stop")
+//            .build()
+    )
+
+    private val startStep = PassiveGaitStep("start", asyncActions)
+
+    private val stopStep = PassiveGaitStep("stop", asyncActions)
+
+    private val steps = mutableListOf<Step>(startStep, stopStep)
+
+    override fun onCreate() {
+        super.onCreate()
+        Log.d(TAG, "onCreate: $taskResultManager, $recorderConfigPresentationFactory")
+
+        task = TaskBase.builder()
+                .setIdentifier("PassiveGait")
+                .setAsyncActions(asyncActions)
+                .setSteps(steps)
+                .build()
+        taskIdentifier = "PassiveGait"
+        taskUUID = UUID.randomUUID()
+
+        recorderManager = RecorderManager(task, taskIdentifier, taskUUID, this,
+                taskResultManager, recorderConfigPresentationFactory)
+
+        val singleTaskRResultManagerConn: Single<TaskResultManagerConnection> = taskResultManager.getTaskResultManagerConnection(taskIdentifier, taskUUID)
+        Log.d(TAG, "$singleTaskRResultManagerConn")
+        singleTaskRResultManagerConn
+                .doOnSuccess { conn ->
+                    println("TEST - doOnSuccss: $conn")
+                    conn.asyncResultsObservable.doOnEach { r -> Log.d(TAG, "doOnEach: $r") }
+                }
+                .doOnError { e ->
+                    println("TEST - doOnError: ${e.message}")
+                }
+    }
+
+    private fun taskResultObserver(taskResult: TaskResult?) {
+        Log.d(TAG, "Observed TaskResult: {$taskResult}")
+    }
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        Log.d(TAG, "onStartCommand: startId = $startId, isRecording = $isRecording")
+
+        intent?.let {
+            when (val event = it.getIntExtra(EVENT, -1)) {
+                STARTED_WALKING -> startRecording()
+                STOPPED_WALKING -> stopRecording()
+                else -> throw IllegalArgumentException("EVENT is invalid: $event")
```

Comment:
Could we ever get here?  I'm not sure that i'd want this to crash.

Reply:
will remove.